core/src/main/java/org/frankframework/receivers/Receiver.java

```diff
@@ -1403,6 +1408,36 @@ private Message processMessageInAdapter(MessageWrapper<M> messageWrapperOriginal
 		}
 	}
 
+	/**
+	 * Last-ditch option to mark the message processing as failure, in case processing fails only at the commit and
+	 * the commit is from a JMS transaction initiated by JMS client / Application Server.
+	 *
+	 * @param messageWrapper Message for which to register the failure.
+	 */
+	private void registerTransactionFailureHandler(MessageWrapper<M> messageWrapper) {
+		if (TransactionSynchronizationManager.isSynchronizationActive()) {
+			TransactionSynchronizationManager.registerSynchronization(new TransactionSynchronization() {
+				@Override
+				public void afterCompletion(int status) {
+					if (status != TransactionSynchronization.STATUS_COMMITTED) {
+						log.debug("{} after rollback, messageId [{}]", Receiver.this::getLogPrefix, messageWrapper::getId);
+						ProcessStatusCacheItem cachedProcessStatus = getCachedProcessStatus(messageWrapper);
+						if (cachedProcessStatus.exitState == ExitState.SUCCESS) {
+							// We thought the message was a success but now it turns out to be an error after all
+							cachedProcessStatus.exitState = ExitState.ERROR;
+							String comment = cachedProcessStatus.comments;
+							cachedProcessStatus.comments = "Error in transaction commit; rollback after successful processing" + (comment == null ? "" : "; " + comment);
+
+							error("Message appeared to have been processed successfully but transaction rolled back unexpectedly", null);
+							getAdapter().incNumOfMessagesInError();
```

Comment:
Moet de message processed dan nog omlaag?

Reply:
Voor zover ik in de code kan nagaan niet, ook berichten in error worden geteld bij "processed".